APEXMALHAR-2380 Add MutablePair for Kinensis Operator for Recovery State

deepak-narkhede opened the pull request:
Add MutablePair for Kinensis Operator for Recovery State which will avoid default serialization caused by KinesisPair.

Tested with Kinesis to S3 app.

chaithu14 commented on contrib/src/main/java/com/datatorrent/contrib/kinesis/AbstractKinesisInputOperator.java:
```diff
@@ -58,16 +32,27 @@
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
```
Why are you changing the import order?

deepak-narkhede replied:
@chaithu14 . It was not intentional, IDE did some import optimizations. I've reverted them.

chaithu14 commented on contrib/src/main/java/com/datatorrent/contrib/kinesis/AbstractKinesisInputOperator.java:
```diff
@@ -565,11 +554,11 @@ public void emitTuples()
       if(!currentWindowRecoveryState.containsKey(shardId))
       {
-        currentWindowRecoveryState.put(shardId, new KinesisPair<String, Integer>(recordId, 1));
+        currentWindowRecoveryState.put(shardId, new MutablePair<String, Integer>(recordId, 1));
```
I would suggest optimize this if block as below:
```java
MutablePair<String, Integer> shardOffsetAndCount = currentWindowRecoveryState.get(shardId);
if (shardOffsetAndCount == null) {
  currentWindowRecoveryState.put(shardId, new MutablePair<String, Integer>(recordId, 1));
} else {
  shardOffsetAndCount.setRight(shardOffsetAndCount.right+1);
}
```

deepak-narkhede replied:
@chaithu14 . Thanks, Done.

deepak-narkhede committed:
APEXMALHAR-2380 Add MutablePair for Kinensis Operator for Recovery State

Conflicts:
	contrib/src/main/java/com/datatorrent/contrib/kinesis/AbstractKinesisInputOperator.java
